src/plugins/janus_videoroom.c

```diff
@@ -3111,10 +3117,12 @@ static janus_videoroom_subscriber_stream *janus_videoroom_subscriber_stream_add_
 		}
 		temp = temp->next;
 	}
-	if(found)
+	if(found)  {
+		stream->opaqueid = g_strdup(opaqueid);
```

Comment:
See above. Besides, you're introducing leaks if `stream->opaqueid` exists already, as you're replacing the pointer and the old one will never be freed.